src/models/Call.ts

```diff
@@ -980,27 +981,45 @@ export class ElementCall extends Call {
         this.participants = participants;
     }
 
+    private onDeviceMute = (ev: CustomEvent<IWidgetApiRequest>): void => {
+        ev.preventDefault();
+        this.messaging!.transport.reply(ev.detail, {}); // ack
+    };
+
     private onHangup = async (ev: CustomEvent<IWidgetApiRequest>): Promise<void> => {
         ev.preventDefault();
-        await this.messaging!.transport.reply(ev.detail, {}); // ack
+        this.messaging!.transport.reply(ev.detail, {}); // ack
+        if (ev.detail.data.cause === "error") {
+            // Switch back to the call view to show the error message
+            defaultDispatcher.dispatch<ViewRoomPayload>({
+                action: Action.ViewRoom,
+                room_id: this.roomId,
+                view_call: true,
+                metricsTrigger: undefined,
+            });
+        }
         this.setDisconnected();
         // In video rooms we immediately want to reconnect after hangup
         // This starts the lobby again and connects to all signals from EC.
         if (isVideoRoom(this.room)) {
             this.start();
+        } else if (ev.detail.data.cause === "user") {
+            // User is done with the call; hide it
+            this.presented = false;
+            this.checkDestroy();
         }
     };
 
     private onTileLayout = async (ev: CustomEvent<IWidgetApiRequest>): Promise<void> => {
         ev.preventDefault();
         this.layout = Layout.Tile;
-        await this.messaging!.transport.reply(ev.detail, {}); // ack
+        this.messaging!.transport.reply(ev.detail, {}); // ack
     };
 
     private onSpotlightLayout = async (ev: CustomEvent<IWidgetApiRequest>): Promise<void> => {
         ev.preventDefault();
         this.layout = Layout.Spotlight;
-        await this.messaging!.transport.reply(ev.detail, {}); // ack
+        this.messaging!.transport.reply(ev.detail, {}); // ack
     };
 
```

Comment:
Do you think this is the right place to start deprecating those actions?
It seems design and product have moved away from this on all platforms for some time.
(Or do we still use  `this.layout`)

I think a first step could be to deprecate the this.layout field.

Reply:
Oh, I see now that the switch has been moved out of the header in https://www.figma.com/design/NKjLzSlbD8JnldYQ5sFpce/Element-Call-Updates-(New)---Q1-2025?node-id=243-16166&p=f&t=rwZYJ2V8QQ5kM0Rp-0. I can remove this stuff in a follow-up PR.